### PR TITLE
Randomized percent on load

### PR DESCRIPTION
**Pull request by asimon-1**
Tldr: Allows you to apply a random damage between a specified minimum and maximum to either the player or CPU when loading a save state.

List of changes:
1. Replaced the "Save Damage" submenu with "Save Dmg (CPU)" and "Save Dmg (Player)"
2. Changed the type of these menus to be a new enum SaveDamage, with options Default/Saved/Random
3. Refactor saved damage implementation to account for player and for "Random" option
4. Added two new submenus "Dmg Range (CPU)" and "Dmg Range (Player)"
5. Added serialization/deserialization for slider submenu types
6. Added panic handler to allow easier debugging
7. Added web component for slider submenu type
8. Added TUI component for slider submenu type
9. Ensured that TUI is not wrapped due to size limitations
10. Paginated TUI tabs to work around wrap change
11. Fixed TUI local rendering using `cargo run --features has_terminal`
12. Made a new tab for Save State options and moved relevant submenus to it

### CODE REVIEW

File: training_mod_tui/src/lib.rs [hunk 1]
```diff
@@ -1,367 +1,806 @@
...
+            .unwrap()
+            .idx_to_list_idx(self.menu_items.get(self.tab_selected()).unwrap().state);
+        let selected_sub_menu = &self.menu_items.get(self.tab_selected()).unwrap().lists
+            [list_section]
+            .items
+            .get(list_idx)
+            .unwrap();
+
+        let toggles = selected_sub_menu.toggles.clone();
+        let _slider = selected_sub_menu.slider.clone();
```
Comment (jugeeya) [inline training_mod_tui/src/lib.rs:72]: super nit: since we use it, shouldn't have `_` prefix

File: training_mod_tui/src/lib.rs [hunk 2]
```diff
@@ -1,367 +1,806 @@
...
+                    toggles,
+                    if selected_sub_menu.toggles.len() >= 3 {
+                        3
+                    } else {
+                        selected_sub_menu.toggles.len()
+                    },
+                )
+            }
+            SubMenuType::SLIDER => {
+                let slider = _slider.unwrap();
```
Comment (jugeeya) [inline training_mod_tui/src/lib.rs:85]: this can be shadowed i think? `let slider = slider.unwrap();`

File: training_mod_tui/src/lib.rs [hunk 3]
```diff
@@ -1,367 +1,806 @@
...
+                }
+            },
+        )
+    }
+
+    /// Returns information about the currently selected slider
+    /// 0: Title
+    /// 1: Help text
+    /// 2: Reference to self.selected_sub_menu_slider
+    pub fn sub_menu_strs_for_slider(&mut self) -> (&str, &str, &DoubleEndedGauge) {
```
Comment (jugeeya) [inline training_mod_tui/src/lib.rs:255]: I'd be down if we have a struct for this output rather than tuple return like `SliderInfo`, similar for toggles. Wonder about your thoughts.

Comment (asimon-1) [reply training_mod_tui/src/lib.rs:255]: Agree that it'd be nice to have in a struct, but I'm going to punt this for later. I'll add a TODO for a cleanup effort.

**Review by jugeeya — APPROVED** [review]
